### PR TITLE
travis: PHP 5.5, 5.6 and 7.0 nightly added

### PR DESCRIPTION
**Pull request by TomasVotruba**
Ref https://github.com/SimpleBus/MessageBus/pull/20

### CODE REVIEW

**Comment by matthiasnoback** [comment]
@TomasVotruba Could you take a look at this? It's currently failing.

**Comment by TomasVotruba** [comment]
Quote typo, now all works fine.

**Comment by TomasVotruba** [comment]
Btw, I'd suggest dropping PHPUnit from composer, since Travis has it installed locally and it has no added value to download ~20 packages of PHPUnit for every build.

E.g. here it's 120 package downloads with no use.

In case you use PHPStorm, PHPUnit can be included as external library (_File > Settings > Project Settings > PHP > add_ - select separate composer PHPUnit installation > _ok_), to provide IDE suggestion.

File: .travis.yml [hunk 1]
```diff
@@ -3,17 +3,28 @@ language: php
...
+      env: SYMFONY_VERSION="2.5.*"
+    - php: 5.6
+      env: SYMFONY_VERSION="2.6.*"
+    - php: 5.6
+      env: SYMFONY_VERSION="2.7.*@dev"
 
 before_install:
   - composer selfupdate
-  - composer require --no-update symfony/http-kernel:${SYMFONY_VERSION} symfony/dependency-injection:${SYMFONY_VERSION} symfony/yaml:${SYMFONY_VERSION}
+  - sh -c 'if [ "$SYMFONY_VERSION" != "" ]; then composer require --no-update symfony/http-kernel:${SYMFONY_VERSION} symfony/dependency-injection:${SYMFONY_VERSION} symfony/yaml:${SYMFONY_VERSION}; fi;'
```
Comment (xabbuh) [inline .travis.yml]: There's no need to wrap the condition in a call to `sh`.

File: .travis.yml [hunk 2]
```diff
@@ -3,17 +3,28 @@ language: php
...
 
-env:
-  - SYMFONY_VERSION: 2.3.*
-  - SYMFONY_VERSION: 2.4.*
-  - SYMFONY_VERSION: 2.5.*
-  - SYMFONY_VERSION: 2.6.*
-  - SYMFONY_VERSION: 2.7.*@dev
+matrix:
+  allow_failures:
+    - php: 7.0
```
Comment (xabbuh) [inline .travis.yml:12]: I suggest to [allow "fast finish" of Travis builds](http://docs.travis-ci.com/user/build-configuration/#Fast-finishing) too.

File: .travis.yml [hunk 3]
```diff
@@ -3,17 +3,28 @@ language: php
...
+  include:
+    - php: 5.6
+      env: SYMFONY_VERSION="2.3.*"
+    - php: 5.6
+      env: SYMFONY_VERSION="2.4.*"
```
Comment (xabbuh) [inline .travis.yml:17]: Does it make sense to run tests for unmaintained Symfony versions?

File: .travis.yml (hunk 1 above)
Comment (TomasVotruba) [inline .travis.yml]: I have tried this before with same expectation, but `${SYMFONY_VERSION}` does not get resolved as variable.

Comment (xabbuh) [inline .travis.yml]: Using the variable name without curly braces will probably work.

Comment (TomasVotruba) [inline .travis.yml]: Have tried that as well.

Comment (xabbuh) [inline .travis.yml]: Honestly, I was not able to reproduce your issue (see xabbuh/SymfonyBridge@e631b9797be8de71853a13cbc7f39a795f5c5a00 and its [Travis build](https://travis-ci.org/xabbuh/SymfonyBridge/builds/54621952)).

Comment (TomasVotruba) [inline .travis.yml]: You are correct. Thanks for example. It made me try [myself and it works](https://github.com/Zenify/DoctrineMethodsHydrator/commit/48483e3d338201cdaa1c9f26e69286b621891ea3).

I probably mixed up something else. I'll fix it here.

File: .travis.yml (hunk 2 above)
Comment (TomasVotruba) [inline .travis.yml:12]: This PR is mainly about PHP 7. It's been sent to other repos of this organization without this option. Feel free to send new PRs.

**Comment by TomasVotruba** [comment]
Ping @matthiasnoback

**Comment by matthiasnoback** [comment]
Thanks for your work on this!
@xabbuh For as long as it is easy to support Symfony 2.3, I'd like to do it.
@TomasVotruba About including PHPUnit: I like the version to be more or less fixed (it probably means that we should have `composer.lock` included in the project). I want the tests to run in the exact same way locally as on Travis and using Composer for that is (I think) the only option that always works. Other options would require some manual verification.

**Comment by TomasVotruba** [comment]
@matthiasnoback Version note was probably about Symfony 2.4, [that is in EOL since January 2015](http://symfony.com/doc/current/contributing/community/releases.html#schedule).

I understand PHPUnit note, thanks for explaining.

**Comment by matthiasnoback** [comment]
Ah! That makes sense. Well, let's just keep it there for now :)

**Comment by TomasVotruba** [comment]
Ok

**Pull request merged**